src/corehost/cli/apphost/bundle/bundle_runner.cpp

```diff
@@ -302,22 +273,11 @@ StatusCode bundle_runner_t::extract()
 {
     try
     {
-        // Determine if the current executable is a bundle
         reopen_host_for_reading();
 
-        //  If the current AppHost is a bundle, it's layout will be 
-        //    AppHost binary 
-        //    Embedded Files: including the app, its configuration files, 
-        //                    dependencies, and possibly the runtime.
-        //    Bundle Manifest
-
-        int64_t manifest_header_offset;
-
-        if (!process_manifest_footer(manifest_header_offset))
-        {
-            return StatusCode::AppHostExeNotBundle;
-        }
-        process_manifest_header(manifest_header_offset);
+        // Read the bundle header
+        seek(m_bundle_stream, marker_t::header_offset(), SEEK_SET);
```

Comment:
It's often a good idea to replace `seek()`+`read()` with `pread()` if supported by the system.  This way, if reading with an invalid offset, the `pread()` call will fail.

Reply:
@vitek-karas  the `seek()` function above already checks for failure, and throws the bundleIOerror exception.

Reply:
@lpereira  I agree that the PAL can be improved by adding `pread()`. 
But I think this code will soon change to use memory-mapping of the whole bundle file instead of seek-read-write, etc. So, I left it as-is for now.